mkosi/sandbox.py

```diff
@@ -806,7 +817,11 @@ def main() -> None:
     if suppress_chown and (userns or userns_has_single_user()):
         seccomp_suppress_chown()
 
-    unshare(namespaces)
+    try:
+        unshare(namespaces)
+    except OSError as e:
+        if e.errno == EPERM:
+            print(UNSHARE_EPERM_MSG, file=sys.stderr)
 
```

Comment:
Is this really the only one that can fail due to permission issues? Or is this only when apparmor fails? What about if the sysctls are enabled?

Reply:
Just noticed my Ubuntu system has the `kernel.unprivileged_userns_clone` patch, so I tried that out. And indeed if I set that to 0 the failure happens in `become_user`. So I've just duplicated the logic there.

I think this could easily happen in a third place too, we probably can't make this foolproof, even if we just do it inside `unshare`. (E.g. I know of a certain Linux version that "solves" this problem in a third way that is neither `kernel.unprivileged_userns_clone` nor an LSM policy, I dunno where mkosi would fail on that system, I can't easily test it or share the details either).